canonical-web-and-design/maas-ui
# refactor(ui): refactor machine PCI/USB tabs to work with controllers

## Done

- Refactored machine PCI/USB tabs to work with controllers

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the PCI and USB tabs of a machine and check that they are unchanged
- Go to the PCI and USB tabs of a controller and check that they work as expected

## Fixes

Fixes canonical-web-and-design/app-tribe#948
Fixes canonical-web-and-design/app-tribe#949

## Screenshot

![Screenshot 2022-06-02 at 11-54-36 bolla PCI devices ⚠️ bolla MAAS](https://user-images.githubusercontent.com/25733845/171530555-57bcbbde-d777-44c3-914e-327bd3d1bded.png)
